lnbits/wallets/corelightningrest.py

```diff
@@ -0,0 +1,241 @@
+    def __init__(self):
+        macaroon = settings.corelightning_rest_macaroon
+        assert macaroon, "missing cln-rest macaroon"
+
+        self.macaroon = load_macaroon(macaroon)
+
+        url = settings.corelightning_rest_url
+        if not url or not macaroon:
+            raise Exception("cannot initialize corelightning-rest")
+
+        self.url = url[:-1] if url.endswith("/") else url
+        self.auth = {
+            "macaroon": self.macaroon,
+            "encodingtype": "hex",
+            "accept": "application/json",
+        }
+
+        self.cert = settings.corelightning_rest_cert or False
+        self.client = httpx.AsyncClient(verify=self.cert, headers=self.auth)
+        self.last_pay_index = 0
+        self.statuses = {
+            "paid": True,
+            "complete": True,
+            "failed": False,
+            "pending": None,
+        }
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        r = await self.client.get(f"{self.url}/v1/channel/localremotebal", timeout=5)
+
+        if r.is_error or "error" in r.json():
+            try:
+                data = r.json()
+                error_message = data["error"]
+            except Exception:
+                error_message = r.text
+            return StatusResponse(
+                f"Failed to connect to {self.url}, got: '{error_message}...'", 0
+            )
+
+        data = r.json()
+        if len(data) == 0:
+            return StatusResponse("no data", 0)
+
+        return StatusResponse(None, int(data.get("localBalance") * 1000))
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        label = f"lbl{random.random()}"
+        data: Dict = {
+            "amount": amount * 1000,
+            "description": memo,
+            "label": label,
+        }
+        if description_hash and not unhashed_description:
+            raise Unsupported(
+                "'description_hash' unsupported by CoreLightningRest, "
+                "provide 'unhashed_description'"
+            )
+
+        if unhashed_description:
+            data["description"] = unhashed_description.decode("utf-8")
+
+        if kwargs.get("expiry"):
+            data["expiry"] = kwargs["expiry"]
+
+        if kwargs.get("preimage"):
+            data["preimage"] = kwargs["preimage"]
+
+        r = await self.client.post(
+            f"{self.url}/v1/invoice/genInvoice",
+            data=data,
+            timeout=40,
+        )
+
+        if r.is_error or "error" in r.json():
+            try:
+                data = r.json()
+                error_message = data["error"]
```

Comment:
- it is possible for `data` not to have an  `"error"` field (`or` condition above)